enarx-keep-sgx-shim/src/start.S

```diff
@@ -163,6 +196,28 @@ syscall:
 
     xor     %rcx,                   %rcx            # Clear %rcx
     zerof   %rcx                                    # Clear CPU flags
+
+    # Save stack pointer, allocate space, and set
+    # 64-byte alignment on stack for xsavec
+    mov     %rsp,                   %rbx
+    sub     $4096,                  %rsp
+    and     $(-0x40),               %rsp
+
+    # Zero the xsave header area on stack
+    movq    $0,        512(%rsp)
+    movq    $0,        520(%rsp)
+    movq    $0,        528(%rsp)
+    movq    $0,        536(%rsp)
+    movq    $0,        544(%rsp)
+    movq    $0,        552(%rsp)
+    movq    $0,        560(%rsp)
+    movq    $0,        568(%rsp)
+
+    # Set mask and perform xsavec
+    movq    $~0,                   %rdx
+    movq    $~0,                   %rax
+    xsavec (%rsp)
```

Comment:
```suggestion
    xsavec (%rsp)
    pushq  %rbx
```

Reply:
save and restore %rbx ... otherwise it will be overriden by the non-enclave-rbx

Reply:
@haraldh Thanks. I was trying to think through which (if any) registers are preserved when we do a syscall. `%rbx` is callee-saved, but I guess we don't trust any "saved" info from the host anyway.

Reply:
@lkatalin There are no preserved registers across a trust boundary. Calling conventions work because you trust the thing you call not to do what it isn't supposed to do. We can't presume that in this case.

Reply:
@npmccallum Thanks, that makes more sense the more I think about it.

Reply:
@lkatalin Anything that you want to save needs to be pushed onto the trusted stack before exit and then popped off the trusted stack when you return.